contrib/guix/libexec/build.sh

```diff
@@ -133,18 +133,7 @@ for p in "${PATHS[@]}"; do
 done
 
 # Disable Guix ld auto-rpath behavior
-case "$HOST" in
-    *darwin*)
-        # The auto-rpath behavior is necessary for darwin builds as some native
-        # tools built by depends refer to and depend on Guix-built native
-        # libraries
-        #
-        # After the native packages in depends are built, the ld wrapper should
-        # no longer affect our build, as clang would instead reach for
-        # x86_64-apple-darwin-ld from cctools
-        ;;
-    *) export GUIX_LD_WRAPPER_DISABLE_RPATH=yes ;;
-esac
+export GUIX_LD_WRAPPER_DISABLE_RPATH=yes
```

Comment:
Why?

Reply:
Why are we setting this, or why are we dropping it for Darwin?

We no-longer need it given we aren't building Darwin tools. It's been disabled for all other hosts since Guix was introduced. I can dig up the history and post it here.

Reply:
Ah, sorry, I misread. Had it backwards. It looked like it was being applied only to darwin before, but it was applied to everything _but_ darwin before. Now darwin is no longer an exception.

Thanks.